lib/remote/jsonrpcconnection-pki.cpp

```diff
@@ -207,23 +196,9 @@ Value RequestCertificateHandler(const MessageOrigin::Ptr& origin, const Dictiona
 		}
 	}
 
-	pubkey = std::shared_ptr<EVP_PKEY>(X509_get_pubkey(cert.get()), EVP_PKEY_free);
-	subject = X509_get_subject_name(cert.get());
-
-	newcert = CreateCertIcingaCA(pubkey.get(), subject);
-
-	/* verify that the new cert matches the CA we're using for the ApiListener;
-	 * this ensures that the CA we have in /var/lib/icinga2/ca matches the one
-	 * we're using for cluster connections (there's no point in sending a client
-	 * a certificate it wouldn't be able to use to connect to us anyway) */
-	try {
-		if (!VerifyCertificate(cacert, newcert, listener->GetCrlPath())) {
-			Log(LogWarning, "JsonRpcConnection")
-				<< "The CA in '" << listener->GetDefaultCaPath() << "' does not match the CA which Icinga uses "
-				<< "for its own cluster connections. This is most likely a configuration problem.";
-			goto delayed_request;
-		}
-	} catch (const std::exception&) { } /* Swallow the exception on purpose, cacert will never be a non-CA certificate. */
+	if (!(newcert = listener->RenewCert(cert))) {
```

Comment:
I mean especially here.

```cpp
std::shared_ptr<X509> newcert;
if (!(newcert = listener->RenewCert(cert))) {
```

isn't even shorter than

```cpp
std::shared_ptr<X509> newcert = listener->RenewCert(cert);
if (!newcert) {
```

Reply:
GHA says: no. :)

Reply:
Well, it says that the variable declaration can't be moved down here (you have to love `goto`...). Still, no harm to just doing a regular assignment outside of the condition.